x-pack/plugins/maps/public/components/layer_addpanel/view.js

```diff
@@ -50,17 +62,17 @@ export class AddLayerPanel extends Component {
       return null;
     }
 
-    const { layerLoading, temporaryLayers, nextAction } = this.props;
+    const { selectLayerAndAdd } = this.props;
+    const { hasLayerSelected, isLoading } = this.state;
```

Comment:
Access these from the props directly. no need to copy these first to the component-state. (see other comment). also, this destructuring isn't really adding anything.